common/scala/src/main/scala/whisk/common/RingBuffer.scala

```diff
@@ -0,0 +1,16 @@
+package whisk.common
+
+import org.apache.commons.collections.buffer.CircularFifoBuffer
+
+object RingBuffer {
+    def apply[T](size: Int) = new RingBuffer[T](size)
+}
+
+class RingBuffer[T](size: Int) {
+    private val inner = new CircularFifoBuffer(size)
+
+    def add(el: T) = inner.add(el)
+    def count(el: T) = {
+        inner.toArray().asInstanceOf[Array[T]].count(_ == el)
+    }
```

Comment:
The `count` call was always when an ActiveAck came back to the invoker.
Now I changed the code, that the count is not called when the Invoker is already Healthy and the state of the activation was successful: https://github.com/openwhisk/openwhisk/pull/1922/commits/dd9db127c2d5973e2c0287123ca220ba4d829358#diff-d2e5bc035bf23b1b663aceb699f5f6e0L288